contrib/recastdemomod/Source/Filelist.cpp

```diff
@@ -55,10 +102,15 @@ void scanDirectory(const char* path, const char* ext, FileList& list)
 	while ((current = readdir(dp)) != 0)
 	{
 		int len = strlen(current->d_name);
-		if (len > 4 && strncmp(current->d_name+len-4, ext, 4) == 0) // TODO:: may not work due to joker(*) added
```

Comment:
This change stands out to me, what is its purpose, and have you validated it works on windows

Reply:
This is under a platform-specific block for *nix, so it shouldn't have an impact.
In CMaNGOS_Map::ScanFoldersForMaps, this function is called like this:
`scanDirectory(m_ctx->getMapFolder(), "*.map", m_MapFiles);`
The issue is that we are doing a string comparison on the file suffix (ex: "0011246.map" -> ".map") and a pattern ("*.map"). Needed to offset pattern by 1 to compare ".map" to ".map" instead of "*.map"

Separately, added an else condition for more complicated pattern matching inside CMaNGOS_Map::scanFoldersForTiles() where we need to load tiles where the file has both prefix and suffix (ex: "001*.mmtile")